bottom-sheet-delegate/BottomSheet/BottomSheetViewPresenter.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  Copyright © FINN AS. All rights reserved.
+//
+
+import UIKit
+
+protocol BottomSheetViewPresenterDelegate: AnyObject {
+    func bottomSheetViewPresenter(_: BottomSheetViewPresenter, didTransitionTo state: BottomSheetState?)
+}
+
+final class BottomSheetViewPresenter {
+    weak var delegate: BottomSheetViewPresenterDelegate?
+
+    // MARK: - Private properties
+
+    private(set) var state: BottomSheetState = .compact
+    private var models: [BottomSheetState: BottomSheetModel] = [:]
+    private var topConstraint: NSLayoutConstraint!
+    private weak var containerView: UIView?
+
+    private lazy var panGesture = UIPanGestureRecognizer(
+        target: self,
+        action: #selector(handlePan(panGesture:))
+    )
+
+    private lazy var springAnimator = SpringAnimator(
+        dampingRatio: 0.8,
+        frequencyResponse: 0.4
+    )
+
+    // MARK: - Internal methods
+
+    public func addPresentedView(_ presentedView: UIView, to containerView: UIView) {
+        guard let compactModel = models[.compact] else { return }
+
+        self.containerView = containerView
+
+        let bottomSheetView = BottomSheetView(contentView: presentedView)
+        bottomSheetView.translatesAutoresizingMaskIntoConstraints = false
+        containerView.addSubview(bottomSheetView)
+
+        topConstraint = bottomSheetView.topAnchor.constraint(
+            equalTo: containerView.topAnchor,
+            constant: containerView.frame.maxY
+        )
+
+        NSLayoutConstraint.activate([
+            topConstraint,
+            bottomSheetView.bottomAnchor.constraint(greaterThanOrEqualTo: containerView.bottomAnchor),
+            bottomSheetView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+            bottomSheetView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor),
+            bottomSheetView.heightAnchor.constraint(greaterThanOrEqualToConstant: compactModel.height)
```

Comment:
With this constraint the `containerView`'s height will be `compactModel.height - 20`. That is a bit annoying with the current `BottomSheet` because you need to add the height of the handle. Maybe you can add a static constant with the height of the handle and add it here?